### DISCUSSION TITLE
stylus re-maintain plan

### DISCUSSION THREAD

**Discussion by iChenLei** [stylus/stylus#2637] [General]
<p align="center"><a href="https://stylus-lang.com" target="_blank" rel="noopener noreferrer"><img width="150" src="https://raw.githubusercontent.com/stylus/stylus/dev/graphics/Logos/stylus.png" alt="Stylus logo"></a></p>

### Hi, all

  I create this discussion for current stylus collaborator `groenroos` , `ota-meshi` and `ThisIsManta`. It's a simple informal team meeting, so the first thing is that introducing the team members.

|  member  |  name  | stylus oss background |
|:--|:--|:--|
| ![ota-meshi](https://avatars.githubusercontent.com/u/16508807?v=4&s=80) | [ota-meshi](https://github.com/ota-meshi)(Yosuke Ota) | [stylelint-plugin-stylus](https://github.com/ota-meshi/stylelint-plugin-stylus) ![NPM DM](https://img.shields.io/npm/dm/stylelint-plugin-stylus) , <br /> [postcss-styl](https://github.com/ota-meshi/postcss-styl) ![NPM downloads](https://img.shields.io/npm/dm/postcss-styl) |
| ![ThisIsManta](https://avatars.githubusercontent.com/u/5592654?v=4&s=80) | [ThisIsManta](https://github.com/ThisIsManta)(Manta Anantachai S.) | [vscode-stylus-supremacy](https://github.com/ThisIsManta/vscode-stylus-supremacy) ![supremacy](https://img.shields.io/visual-studio-marketplace/i/thisismanta.stylus-supremacy) , <br /> [stylus-supremacy](https://github.com/ThisIsManta/stylus-supremacy) ![NPM](https://img.shields.io/npm/dm/stylus-supremacy)|
| ![groenroos](https://avatars.githubusercontent.com/u/951547?v=4&s=80) | [groenroos](https://github.com/groenroos)(Oskari Groenroos) | [bulma-stylus](https://github.com/groenroos/bulma-stylus) ![NPM](https://img.shields.io/npm/dm/bulma-stylus), <br /> [Stylus PR](https://github.com/stylus/stylus/commits?author=groenroos) |
| ![iChenLei](https://avatars.githubusercontent.com/u/14012511?v=4&s=80) | [iChenLei](https://github.com/iChenLei)(Lei Chen) | [vscode-stylus](https://github.com/d4rkr00t/language-stylus) ![stylus](https://img.shields.io/visual-studio-marketplace/i/sysoev.language-stylus) , [Stylus PR](https://github.com/stylus/stylus/commits?author=iChenLei) |

Maybe you guys don't know each other, so it's time to know we have a team. :tada:

### Team Daily Communication
- [ ] gitter
- [ ] slack
- [x] discord
- [ ] others
> All team members, please choice one or add your other choices

### Maintenance Roadmap

| Field |  maintainer |  content |  status |
| :--- | :--- | :--- | :--- |
| stylus | All team members | ![issues](https://img.shields.io/github/issues/stylus/stylus) ,  and missing many modern CSS feature support (e.g. [Media Queries Level 4](https://www.w3.org/TR/mediaqueries-4), [css-nesting](https://drafts.csswg.org/css-nesting/), [Selectors Level 4](https://www.w3.org/TR/selectors-4) and more ) | working in progress |
| language server protocol | All team members | for better IDE/editor code intellisense | Under Planning |
| stylus(browser build) | iChenLei |  publish browser build for `unpkg` and other cdn  | working in progress |
| doc | iChenLei & groenroos | stylus-lang.com missing many modern feature, such as `search`, `REPL` and more, a better documentation website is needed. [relative issue comment ](https://github.com/stylus/stylus-lang.com/pull/36#issuecomment-949302263) | Under Planning |
| vscode intergration | iChenLei & ThisIsManta | vscode-stylus and vscode-stylus-supremacy  |  working in progress  |
| stylelint intergration | ota-meshi |  stylelint-plugin-stylus , postcss-styl  |  working in progress  |
| prettier intergration | N/A | prettier format support for `.styl` file | Under Planning |

> Missing something ? Feel free to modify this table

### Github Projects ?

Do we need [github projects](https://github.com/stylus/stylus/projects) to manage our all stylus maintenance work ? Like this:

<img src="https://user-images.githubusercontent.com/14012511/139395842-64b57cf2-3f66-4bb8-bb7c-44bc16b02272.png" width="799px" />

### Repository Migration

| from | to |
| :-- | :-- |
| https://github.com/ThisIsManta/stylus-supremacy | https://github.com/stylus/stylus-supremacy |
| https://github.com/ThisIsManta/vscode-stylus-supremacy | https://github.com/stylus/vscode-stylus-supremacy |
| https://github.com/ota-meshi/stylelint-plugin-stylus | https://github.com/stylus/stylelint-plugin-stylus |
| https://github.com/ota-meshi/postcss-styl | https://github.com/stylus/postcss-styl |
|  more commuity repo | to official repo for better collaborate |

> But all repositories migration need repository owner's agree, it is not mandatory.

cc @ThisIsManta @groenroos @ota-meshi 

> feel free add your comment or modify content directly, thanks, all team members !

**Comment by ThisIsManta** [comment]
Hi guys

Slack or discord is fine by me.
GitHub project looks okay.
I'm cool with repo migration. Feel free to suggest different names for Supremacy if you'd like.

Please note that I have a full time job so my contribution here is somewhat limited.
Cheers!

**Comment by ota-meshi** [comment]
Hi everyone. I'm glad to be a member of this team!
I agree with the transfer of the repos. Please let me know if you have any necessary steps.

**Reply by iChenLei** [reply]
Mr.Ota , Is [Discord](https://discord.com/open-source) ok for you ?

**Reply by ota-meshi** [reply]
Yes.  Discord is good for me.

**Comment by groenroos** [comment]
It's great to meet you guys, and great to welcome new repos into the official organization!

Although when we move them, maybe we can also slightly rename the repos to remove repetition and look more clean, for example `stylus/supremacy` instead of `stylus/stylus-supremacy`?

Discord is good for me.

**Reply by iChenLei** [reply]
Agree with your suggestion for shorter name.
